### PR TITLE
Initial change to watch named input resources

### PR DESCRIPTION
**Pull request by stuggi**
This adds the following fields to index to be able to watch them when they change and the controller should reconcile.
- `.spec.secret`
- `.spec.tls.caBundleSecretName`
- `.spec.tls.ingerss.ca.customIssuer`
- `.spec.tls.podLevel.internal.ca.customIssuer`
- `.spec.tls.podLevel.ovn.ca.customIssuer`

### CODE REVIEW

File: controllers/core/openstackcontrolplane_controller.go [hunk 1]
```diff
@@ -339,8 +346,85 @@ func (r *OpenStackControlPlaneReconciler) reconcileNormal(ctx context.Context, i
...
+	}
+
+	// index tlsPodLevelOvnCACustomIssuer
+	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1beta1.OpenStackControlPlane{}, tlsPodLevelOvnCACustomIssuer, func(rawObj client.Object) []string {
+		// Extract the secret name from the spec, if one is provided
+		cr := rawObj.(*corev1beta1.OpenStackControlPlane)
+		if cr.Spec.TLS.PodLevel.Ovn.Ca.CustomIssuer == nil {
+			return nil
+		}
+		return []string{*cr.Spec.TLS.PodLevel.Ovn.Ca.CustomIssuer}
```
Comment (stuggi) [inline controllers/core/openstackcontrolplane_controller.go:423]: need to fix that

**Review by abays — APPROVED** [review]
/lgtm

**Comment by stuggi** [comment]
damn pushed to the wrong branch ... will revert

**Review by olliewalsh — APPROVED** [review]
/lgtm

**Review by abays — APPROVED** [review]
/lgtm

**Pull request merged**